src/Config.php

```diff
@@ -192,15 +212,18 @@ public function isAlias(string $name): bool
      * {@inheritDoc}
      *
      * @see ConfigInterface::getConfiguredTypeNames()
+     *
+     * @return list<string>
      */
     public function getConfiguredTypeNames(): array
     {
-        return array_keys($this->types);
+        return array_map('strval', array_keys($this->types));
     }
 
     public function setTypePreference(string $type, string $preference, ?string $context = null): self
     {
         if ($context) {
+            /** @psalm-suppress MixedArrayAssignment TODO: Eliminate array structures with the next releases */
             $this->types[$context]['preferences'][$type] = $preference;
```

Comment:
This is something I'd like to eliminate completely in future versions of this component.
All structures should be explicitly shaped and typed. Config elements should be immutable value objects.